segmentation_labeling_app/rois/rois.py

```diff
@@ -0,0 +1,241 @@
+        Args:
+            threshold: The threshold to compare values
+
+        Returns:
+            2D numpy array with values greater than threshold == 1 less than
+            threshold == 0
+        Notes:
+            This function will likely become deprecated as we work on more
+            applicable binary thresh holding
+        """
+        return np.uint8(self._sparse_coo.toarray() > threshold)
+
+    def create_dilated_contour_mask(self, stroke_size: int,
+                                    threshold: float) -> np.array:
+        """
+        Returns a mask for a stroke size and threshold. An edge mask is
+        calculated from the threshold. This mask is then dilated with dilation
+        iterations for all values from 0 to stroke_size. All but the edges
+        are masked away and added to the previous edge mask.
+        This mask is then binarized where each pixel above or equal to 1
+        becomes 1. This produces a border of size stroke_size around the
+        original edge mask.
+        Args:
+            stroke_size: Value in pixels for the size of the stroke
+            threshold: Float to threshold every pixel against to generate
+            binary mask
+
+        Returns:
+            final_mask: the edge binary mask around the ROI where 1 is border
+            and 0 is non border
+        """
+        final_mask = np.zeros(shape=self.image_shape)
+        edge_mask = self.get_edge_mask(threshold=threshold)
+        dilation_kernel = cv2.getStructuringElement(cv2.MORPH_RECT,
+                                                    ksize=(3, 3))
+        for i in range(0, stroke_size):
+            dilated_mask = cv2.dilate(edge_mask, dilation_kernel,
+                                      iterations=i)
+            dilated_mask = self.get_edge_mask(threshold=1,
+                                              optional_array=dilated_mask)
+            final_mask = final_mask + dilated_mask
+        final_mask = np.where(final_mask >= 1, 1, 0)
+        return final_mask
+
+    def get_edge_points(self, threshold: float,
+                        optional_array: np.array = None):
+        """
+        Returns a list of edge points. Computes edges using binary erosion and
+        exclusive or operation. Generates edges from a thresholded binary mask.
+        Args:
+            threshold: a float to threshold all values against to create binary
+                       mask
+            optional_array: an optional array to overload the function and
+            create edge points from different mask
+        Returns:
+            Contours: a list of coordinates that contain edge points
+        """
+        if optional_array is None:
+            optional_array = self.generate_binary_mask_from_threshold(threshold)
+        normalized_array = np.uint8(optional_array * 255)
+        contours, _ = cv2.findContours(normalized_array,
+                                       cv2.RETR_TREE,
```

Comment:
Use CV_RETR_LIST over the tree since you're not using the hierarchical information